src/projection_client.rs

```diff
@@ -0,0 +1,629 @@
+                                        version: details.version,
+                                        epoch: details.epoch,
+                                        effective_name: details.effective_name,
+                                        writes_in_progress: details.writes_in_progress,
+                                        reads_in_progress: details.reads_in_progress,
+                                        partitions_cached: details.partitions_cached,
+                                        status: details.status,
+                                        state_reason: details.state_reason,
+                                        name: details.name,
+                                        mode: details.mode,
+                                        position: details.position,
+                                        progress: details.progress,
+                                        last_checkpoint: details.last_checkpoint,
+                                        events_processed_after_restart: details.events_processed_after_restart,
+                                        checkpoint_status: details.checkpoint_status,
+                                        buffered_events: details.buffered_events,
+                                        write_pending_events_after_checkpoint: details.write_pending_events_after_checkpoint,
+                                        write_pending_events_before_checkpoint: details.write_pending_events_before_checkpoint,
+                                    };
+
+                                    yield Ok(details);
+                                    continue;
+                                }
+
+                                break;
+                            }
+                        }
+                    }
+                };
+
+                let stream: BoxStream<crate::Result<StatsDetails>> = Box::pin(stream);
+
+                Ok(stream)
+            })
+            .await
+    }
+
+    pub async fn enable_projection<Name>(
+        &self,
+        name: Name,
+        credentials: Option<&Credentials>,
+    ) -> crate::Result<()>
+    where
+        Name: AsRef<str>,
+    {
+        let options = projections::enable_req::Options {
+            name: name.as_ref().to_string(),
+        };
+
+        let req = projections::EnableReq {
+            options: Some(options),
+        };
+
+        let mut req = tonic::Request::new(req);
+
+        crate::commands::configure_auth_req(&mut req, credentials.cloned());
+
+        self.client
+            .execute(|handle| async move {
+                let mut client =
+                    projections::projections_client::ProjectionsClient::new(handle.channel.clone());
+
+                let _ = client.enable(req).await?;
+
+                Ok(())
+            })
+            .await
+    }
+
+    pub async fn reset_projection<Name>(
+        &self,
+        name: Name,
+        write_checkpoint: bool,
```

Comment:
**CONCERN:** It seems that on .NET gRPC client, it's not possible to set this param. It's hardcoded to `true`. I'm not sure which version is better, but I believe that we should align that https://github.com/EventStore/EventStore-Client-Dotnet/blob/master/src/EventStore.Client.ProjectionManagement/EventStoreProjectionManagementClient.Control.cs#L39.

Reply:
I personally think the .NET approach is not good. Having less control on that parameter is not the better solution in my opinion.